### PR TITLE
Add e2e tests for Helm support

### PR DESCRIPTION
**Pull request by xinnywinne**
Add more e2e tests to verify `gcenode` and `gcpserviceaccount` auth types work with Artifact Registry.

### CODE REVIEW

File: e2e/testcases/helm_sync_test.go [hunk 1]
```diff
@@ -110,9 +111,11 @@ func TestHelmNamespaceRepo(t *testing.T) {
 		nt.T.Error(err)
 	}
 	// Change the RepoSync to sync from the original git source to make test works in the shared test environment.
-	rs.Spec.SourceType = string(v1beta1.GitSource)
-	nt.RootRepos[configsync.RootSyncName].Add(nomostest.StructuredNSPath(repoSyncNN.Namespace, repoSyncNN.Name), rs)
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Update RepoSync to sync from the original git repository")
+	nt.T.Cleanup(func() {
```
Comment (sdowell) [inline e2e/testcases/helm_sync_test.go:114]: The cleanup should be declared earlier in the test before a failure might occur. In this case it should probably be done when the `SourceType` is changed to use helm

Comment (xinnywinne) [reply e2e/testcases/helm_sync_test.go]: Sounds good, done.

File: e2e/testcases/helm_sync_test.go [hunk 2]
```diff
@@ -172,6 +259,7 @@ func TestHelmARTokenAuth(t *testing.T) {
 		// Change the rs back so that it works in the shared test environment.
 		nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceType": "%s", "helm": null, "git": {"dir": "acme", "branch": "main", "repo": "%s", "auth": "ssh","gcpServiceAccountEmail": "", "secretRef": {"name": "git-creds"}}}}`,
 			v1beta1.GitSource, origRepoURL))
+		nt.MustKubectl("delete", "secret", "foo", "-n", v1.NSConfigManagementSystem, "--ignore-not-found")
```
Comment (sdowell) [inline e2e/testcases/helm_sync_test.go:262]: This can be moved to its own cleanup block that is declared above when the secret is created

Comment (xinnywinne) [reply e2e/testcases/helm_sync_test.go]: Done!

**Pull request merged**